core/src/main/scala/algebra/lattice/GenBool.scala

```diff
@@ -0,0 +1,57 @@
+package algebra
+package lattice
+
+import ring.BoolRng
+import scala.{specialized => sp}
+
+/**
+ * Generalized Boolean algebra, that is, a Boolean algebra without
+ * the top element. Generalized Boolean algebras do not (in general)
+ * have (absolute) complements, but they have ''relative complements''
+ * (see [[GenBool.without]]).
+ */
+trait GenBool[@sp(Int, Long) A] extends Any with DistributiveLattice[A] with BoundedJoinSemilattice[A] { self =>
+  def and(a: A, b: A): A
+  override def meet(a: A, b: A): A = and(a, b)
+
+  def or(a: A, b: A): A
+  override def join(a: A, b: A): A = or(a, b)
+
+  /**
+   * The operation of ''relative complement'', symbolically often denoted
+   * `a\b` (the symbol for set-theoretic difference, which is the
+   * meaning of relative complement in the lattice of sets).
+   */
+  def without(a: A, b: A): A
+
+  /**
+   * Logical exclusive or, set-theoretic symmetric difference.
+   * Defined as `a\b ∨ b\a`.
+   */
+  def xor(a: A, b: A): A = or(without(a, b), without(b, a))
+
+  /**
+   * Every generalized Boolean algebra is also a `BoolRng`, with
+   * multiplication defined as `and` and addition defined as `xor`.
+   */
+  def asBoolRing: BoolRng[A] = new BoolRngFromGenBool(self)
```

Comment:
Actually, this PR is basically done. Let's merge this and not block it on #108. If we remove constructions, this is a marginal amount compared with what we have already.